index.js

```diff
@@ -78,15 +78,28 @@ class S3Grc extends Base {
       cb)
   }
 
-  getDownloadUrl (filename, key, cb = null) {
+  /**
+   * @param {string} filename
+   * @param {string} key
+   * @param {object|function|null} [opts]
+   * @param {string} [opts.bucketName]
+   * @param {function} [cb]
+   * @returns {Promise|void}
+   */
+  getDownloadUrl (filename, key, opts = null, cb = null) {
+    if (typeof opts === 'function') {
+      cb = opts
+      opts = null
+    }
+
     const asciiFileName = getAsciiFileName(filename)
     const responseDisposition = (asciiFileName)
       ? `attachment; filename=${asciiFileName}`
       : 'attachment'
 
     const signedUrlExpireTime = 120
     const s3 = this.conf
-    const bucket = s3.bucket
+    const bucket = (opts && opts.bucketName) || s3.bucket
```

Comment:
here we can use null coalesce alongside optional chaining

Reply:
we should add v16 and above I think since prod env supports it

Reply:
well we don't maintain any versioning on it and node v8 is long deprecated so I would favor on moving it forward but it's not big issue to keep ||, no harm